src/utils.c

```diff
@@ -334,3 +334,54 @@ ts_function_types_equal(Oid left[], Oid right[], int nargs)
 	}
 	return true;
 }
+
+Oid
+get_function_oid(char *name, char *schema_name, int nargs, Oid arg_types[])
+{
+	FuncCandidateList func_candidates;
+
+	func_candidates = FuncnameGetCandidates(list_make2(makeString(schema_name), makeString(name)), nargs, NIL, false, false, false);
+	while (func_candidates != NULL)
+	{
+		if (func_candidates->nargs == nargs && ts_function_types_equal(func_candidates->args, arg_types, nargs))
+			return func_candidates->oid;
+		func_candidates = func_candidates->next;
+	}
+	elog(ERROR, "failed to find function %s in schema %s with %d args", name, schema_name, nargs);
+}
+
+/**
+ * This function will remove target from the list without free-ing allocated memory.
+ * If some parts of the code hold a reference to the list or it's items, they should
+ * continue to work without any problems since their pointers will point to a
+ * valid memory location.
+ * Allocated memory for list and items should be freed by PostgreSQL at the end of
+ * transaction.
+ * NOTE: you should make sure that target belongs to given list
+ */
+List *
+list_cell_detach(List *list, ListCell *target, ListCell *prev)
```

Comment:
This code seems to be identical to list_delete_cell minus the pfree of the ListCell.
Why are you not using list_delete_cell here? Your code leaks the ListCell and list_delete_cell doesn't free the item it only frees the containing ListCell but not the pointers in the ListCell.

Reply:
list item == list cell . Freeing a ListCell (list item) will make an old list invalid.

Reply:
The function code comment clearly states that memory will not be deallocated.

Reply:
Hmmm... haven't we discussed scenario when you save a pointer to the List somewhere in our code that might be used later on? I guess It's the same idea as saving a pointer for a Node so the same principals should apply? Maybe I misunderstood you but in your comment you've talked about freeing the list cell (not Node*)
`Also, is it safe to free the cell (which I believe this does), rather than just detaching it from the list?`

Anyways, I can revert to my old code.

Reply:
Yeah, that's why I said this is partially my fault, because I expressed myself in a sloppy way when asking the question. I should have been more explicit in what I meant by "safe". What I meant was that if freeing the cell also means freeing  the `Node *`, then freeing the cell would be unsafe. That's why I asked whether it is safe to free the cell.